Firestore/src/CollectionReference.php

```diff
@@ -266,6 +269,17 @@ public function listDocuments(array $options = [])
             'mask' => []
         ];
 
+        if (isset($options['readTime'])) {
+            if (!($options['readTime'] instanceof Timestamp)) {
+                throw new \InvalidArgumentException(sprintf(
+                    '`$options.readTime` must be an instance of %s',
+                    Timestamp::class
+                ));
+            }
+
+            $options['readTime'] = $options['readTime']->formatForApi();
+        }
```

Comment:
As this block appears 4 different times across 4 different classes, maybe make a `TimestampTrait` that has a `private` method `formatReadTimeOption` ?

Just a suggestion, not a requirement! But it would more DRY

Reply:
Could you kindly provide me with more information about your suggestion? I'm afraid I didn't quite understand it and would appreciate it if you could explain it to me in more detail. Thank you for your help.